smack-extensions/src/main/java/org/jivesoftware/smackx/bob/element/BoBIQ.java

```
@@ -0,0 +1,108 @@
+
+import java.io.UnsupportedEncodingException;
+
+import org.jivesoftware.smack.packet.IQ;
+import org.jivesoftware.smackx.bob.BoBData;
+import org.jivesoftware.smackx.bob.BoBHash;
+import org.jivesoftware.smackx.bob.BoBManager;
+
+/**
+ * Bits of Binary IQ class.
+ * 
+ * @author Fernando Ramirez
+ * @see <a href="http://xmpp.org/extensions/xep-0231.html">XEP-0231: Bits of
+ *      Binary</a>
+ */
+public class BoBIQ extends IQ {
+
+    /**
+     * data element.
+     */
+    public static final String ELEMENT = "data";
+
+    /**
+     * the IQ NAMESPACE.
+     */
+    public static final String NAMESPACE = BoBManager.NAMESPACE;
+
+    private BoBHash bobHash;
+    private BoBData bobData;
+
+    /**
+     * Bits of Binary IQ constructor.
+     * 
+     * @param bobHash
+     * @param bobData
+     */
+    public BoBIQ(BoBHash bobHash, BoBData bobData) {
+        super(ELEMENT, NAMESPACE);
+        this.bobHash = bobHash;
+        this.bobData = bobData;
+    }
+
+    /**
+     * Bits of Binary IQ constructor.
+     * 
+     * @param bobHash
+     */
+    public BoBIQ(BoBHash bobHash) {
+        this(bobHash, null);
+    }
+
+    /**
+     * Get the BoB hash.
+     * 
+     * @return the BoB hash
+     */
+    public BoBHash getBoBHash() {
+        return bobHash;
+    }
+
+    /**
+     * Get the BoB data.
+     * 
+     * @return the BoB data
+     */
+    public BoBData getBoBData() {
+        return bobData;
+    }
+
+    @Override
+    protected IQChildElementXmlStringBuilder getIQChildElementBuilder(IQChildElementXmlStringBuilder xml) {
+        xml.attribute("cid", bobHash.toCid());
+
+        if (bobData != null) {
+            xml.attribute("max-age", String.valueOf(bobData.getMaxAge()));
+            xml.attribute("type", bobData.getType());
+            xml.rightAngleBracket();
+
+            try {
+                xml.escape(new String(bobData.getContent(), "UTF-8"));
+            } catch (UnsupportedEncodingException e) {
+            }
```

Neva eva swallow exceptions. Never. I think BobData should have a e.g. `getStringContent()` method which does the `byte[]` to `String` transformation and throws an ISE if UTF-8 is not supported.